plugins/kubernetes-backend/src/service/KubernetesBuilder.ts

```diff
@@ -268,6 +311,67 @@ export class KubernetesBuilder {
     const logger = this.env.logger;
     const router = Router();
     router.use(express.json());
+    router.use(
+      createPermissionIntegrationRouter({
+        resourceType: RESOURCE_TYPE_KUBERNETES_RESOURCE,
+        // permissions?: Array<Permission>,
+        rules: [matchesJSON],
+        getResources: (resourceRefs: string[]) => {
+          return Promise.all(
+            resourceRefs.map(async (resourceRef: string) => {
+              // Parse the reference, following a format like this:
+              // "cluster:group/v1beta1.things:namespace/name"
+              const [clusterName, gvk, metadata] = resourceRef.split(':');
+              const [groupVersion, plural] = gvk.split('.');
+              let group: string;
+              let apiVersion: string;
+              if (groupVersion.includes('/')) {
+                [group, apiVersion] = groupVersion.split('/');
+              } else {
+                [group, apiVersion] = ['', groupVersion];
+              }
+              let namespace: string;
+              let name: string;
+              if (metadata.includes('/')) {
+                [namespace, name] = metadata.split('/');
+              } else {
+                [namespace, name] = ['', metadata];
+              }
+
+              // actually grab the referenced resource, naively
+              const clusterDetails = await clusterSupplier
+                .getClusters()
+                .then(clusters => clusters.find(c => c.name === clusterName));
+              if (!clusterDetails) {
+                throw new NotFoundError(`Cluster '${clusterName}' not found`);
+              }
+              if (!clusterDetails.serviceAccountToken) {
+                throw new Error(`No credentials for cluster ${clusterName}`);
+              }
+              const encode = (s: string) => encodeURIComponent(s);
+              let resourcePath = group
```

Comment:
it's around here that I kinda want an interface that very generically takes a cluster name and a request and figures out how to make the given request to the given cluster, appending any necessary auth/TLS